### PR TITLE
redolog: add redo log manager and config

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

part of #2351 redo log based replication to implement eventually consistency

#2685 contains all of the change for the requirement, will split into small ones for review, this is one of the prs, split from the #2685, this one contains #2809,#2810, review #2809,#2810,#2811 first. Or review commit https://github.com/amyangfei/ticdc/commit/6a5fe6bd52f68bf39c35cc699c7d54152b3e0e96 directly

NOTE: This PR is separated

### What is changed and how it works?

- Add redo log manager, maintaining resolved ts of redo log writer of each table.
   - redo log manager interacts with upper caller and redo log writer.
   - In current design, the upper caller is table sink, table sink can emit DMLs and DDLs to redo log manager, redo log manager converts these TiCDC internal events to `redo logs` and pass the redo logs to underlying redo log writer.
      - Note the redo logs are kept in memory, the redo log manager doesn't implement back pressure currently, since the upper caller table sink/sink manager has a built in flow control mechanism.
   - The resolved ts of each table is maintained independently, in order we can implement table based replication in the future.
- Add `consistent` config in changefeed

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/redo/convert.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import (
+	pmodel "github.com/pingcap/parser/model"
+	"github.com/pingcap/ticdc/cdc/model"
+)
+
+// RowToRedo converts row changed event to redo log row
+func RowToRedo(row *model.RowChangedEvent) *model.RedoRowChangedEvent {
+	redoLog := &model.RedoRowChangedEvent{
+		Row:        row,
+		Columns:    make([]*model.RedoColumn, 0, len(row.PreColumns)),
```
Comment (ben1009) [inline cdc/redo/convert.go:25]: should be row.Columns instead ?

File: cdc/redo/manager_test.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	"testing"
+	"time"
+
+	"github.com/pingcap/check"
+	"github.com/pingcap/ticdc/cdc/model"
+	"github.com/pingcap/ticdc/pkg/config"
+	"github.com/pingcap/ticdc/pkg/util/testleak"
+)
+
+func TestSuite(t *testing.T) { check.TestingT(t) }
```
Comment (ben1009) [inline cdc/redo/manager_test.go:27]: switch to testify if possible ?

Comment (amyangfei) [reply cdc/redo/manager_test.go]: changed

File: cdc/redo/convert.go (hunk 1 above)
Comment (amyangfei) [reply cdc/redo/convert.go]: addressed

**Pull request closed without merging**